testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/model/CacheTest.java

```diff
@@ -0,0 +1,193 @@
+import static org.keycloak.testsuite.arquillian.DeploymentTargetModifier.AUTH_SERVER_CURRENT;
+
+
+import org.keycloak.testsuite.AbstractTestRealmKeycloakTest;
+import org.keycloak.testsuite.Assert;
+import org.keycloak.models.utils.ModelToRepresentation;
+import org.keycloak.representations.idm.RealmRepresentation;
+import org.keycloak.testsuite.client.KeycloakTestingClient;
+import org.keycloak.testsuite.arquillian.TestContext;
+
+/**
+ * @author <a href="mailto:bill@burkecentral.com">Bill Burke</a>
+ * @version $Revision: 1 $
+ */
+
+public class CacheTest extends AbstractTestRealmKeycloakTest {
+
+	@Deployment
+	@TargetsContainer(AUTH_SERVER_CURRENT)
+	public static WebArchive deploy() {
+		return RunOnServerDeployment.create(UserResource.class, CacheTest.class)
+				.addPackages(true,
+						"org.keycloak.testsuite",
+						"org.keycloak.testsuite.model");
+	}
+
+
+	private ClientModel testApp = null;
+	private int grantedRolesCount=0;
+	private RealmModel realm = null;
+	private UserModel user = null;
+	
+	 @Override
+	    public void configureTestRealm(RealmRepresentation testRealm) {
+	    }
+
+	 @Test
+	    public void testStaleCache() throws Exception {
+		 testingClient.server().run(session -> {
+		 	String appId = null;
+	        {
+	            // load up cache
+
+	            RealmModel realm = session.realms().getRealmByName("test");
+	            assertTrue(realm instanceof RealmAdapter);
+	            ClientModel testApp = realm.getClientByClientId("test-app");
+	            assertTrue(testApp instanceof ClientAdapter);
+	            assertNotNull(testApp);
+	            appId = testApp.getId();
+	            assertTrue(testApp.isEnabled());
+	     
+	        
+	       
+	            // update realm, then get an AppModel and change it.  The AppModel would not be a cache adapter
+	     
+
+	            // KEYCLOAK-1240 - obtain the realm via session.realms().getRealms()
+	            realm = null;
+	            List<RealmModel> realms = session.realms().getRealms();
+
+	            for (RealmModel current : realms) {
+	                assertTrue(current instanceof RealmAdapter);
+	                if ("test".equals(current.getName())) {
+	                    realm = current;
+	                    break;
+	                }
+	            }
+
+	            realm.setAccessCodeLifespanLogin(200);
+	            testApp = realm.getClientByClientId("test-app");
+
+	            assertNotNull(testApp);
+	            testApp.setEnabled(false);
+	        
+	        
+	        // make sure that app cache was flushed and enabled changed
+	        
+	       
+	            realm = session.realms().getRealmByName("test");
+	            Assert.assertEquals(200, realm.getAccessCodeLifespanLogin());
+	            testApp = session.realms().getClientById(appId, realm);
+	            Assert.assertFalse(testApp.isEnabled());
+	        
+	        }
+		 });
+		 }
+    @Test
+    public void testAddUserNotAddedToCache() {
```

Comment:
Is it possible to use @ModelTest instead and pass KeycloakSession as an argument to the method? Same for other places?

Reply:
I spent too much time writing this file using testingClient and trying to figure out the appropriate patterns for using it.  I'd rather leave it as is, there are no double static calls.